mitodl/micromasters
# Changed OSX Docker workflow to expect the webpack server to be run on the host machine

#### What are the relevant tickets?

#307 

#### What's this PR do?

Changes our workflow so OSX users will run the webpack dev server on their host machines rather than inside the Docker container. This eliminates the need for docker-osx-dev.

#### Where should the reviewer start?

Start at `docker-compose.yml` for the config changes, `micromasters/utils.py` for the code changes

#### How should this be manually tested?

OSX users should run `./webpack_dev_server.sh --install` in one terminal tab, and `docker-compose up`. Linux users can just run the latter command. We'll definitely want a Linux user to test this since I hacked on this in OSX. 

#### Any background context you want to provide?

**Major implications:**

1. (as mentioned) OSX users will need to run the webpack server locally in a separate terminal tab, and can do so by running the helper script `./webpack_dev_server.sh`
1. (as mentioned) docker-osx-dev is no longer needed
1. OSX users will now need to add the following to the `.env` file: `WEBPACK_DEV_SERVER_HOST='localhost'`. Before now, webpack was assumed to be running in the container with the web server.
1. The 'watch' service in `docker-compose.yml` now invokes the helper script to start the server. In the case of OSX users, 'watch' will do nothing. I couldn't figure out a good way to entirely exclude the 'watch' service for OSX users that wouldn't require some annoying flags to be specified every single time the `docker-compose` command was used. Any feedback on that is quite welcome.
